sale_auto_remove_zero_quantity_lines/models/sale_order.py

```diff
@@ -0,0 +1,23 @@
+# Copyright 2024 Camptocamp (<https://www.camptocamp.com>).
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from odoo import _, models
+
+
+class SaleOrder(models.Model):
+    _inherit = "sale.order"
+
+    def action_confirm(self):
+        for order in self:
+            if order.company_id.sale_auto_remove_zero_quantity_lines:
+                zero_lines = order.order_line.filtered(
+                    lambda line: line.product_uom_qty == 0
+                )
+                if zero_lines:
+                    body = _(
```

Comment:
I would have logged the lines names and quantities

Reply:
In our requirements we prefer to not have this level of details being logged in the form.